Makefile

```
@@ -1,11 +1,56 @@
-BUILD_DIR:=./bin
+BUILDFLAGS ?=
+unexport GOFLAGS
 
-build: clean
-	mkdir -p ${BUILD_DIR}
-	go build -o "${BUILD_DIR}/backplane-tools" main.go
+BASE_DIR=$(shell pwd)
+BIN_DIR=${BASE_DIR}/bin
 
-clean:
-	rm -rf "${BUILD_DIR}"
+all: format mod build test lint
 
+format: vet mod fmt mockgen ci-build docs
+
+fmt:
```

```suggestion
.PHONY: fmt
fmt:
```

It's also a bit confusing we have `format` and `fmt` as targets, maybe we could find more different names better describing what the targets do?

---

Ended up removing the `format` target since, after cleaning up it's dependencies, very little was actually accomplished with it. The `vet` and `fmt` dependencies that are still relevant to backplane-tools are now called by `all`.